dojo/tools/sysdig_reports/parser.py

```diff
@@ -11,48 +12,124 @@ class SysdigReportsParser(object):
     """
 
     def get_scan_types(self):
-        return ["Sysdig Vulnerability Report - Pipeline, Registry and Runtime (CSV)"]
+        return ["Sysdig Vulnerability Report"]
 
     def get_label_for_scan_types(self, scan_type):
         return "Sysdig Vulnerability Report Scan"
 
     def get_description_for_scan_types(self, scan_type):
-        return "Import of Sysdig Pipeline, Registry and Runtime Vulnerability Report Scans in CSV format."
+        return "Import of Sysdig Pipeline, Registry and Runtime Vulnerability Report Scans in CSV format or a Sysdig UI JSON Report"
 
     def get_findings(self, filename, test):
-
         if filename is None:
             return ()
-
         if filename.name.lower().endswith('.csv'):
             arr_data = CSVParser().parse(filename=filename)
+            return self.parse_csv(arr_data=arr_data, test=test)
+        elif filename.name.lower().endswith('.json'):
+            scan_data = filename.read()
+            try:
+                data = json.loads(str(scan_data, "utf-8"))
+            except Exception:
+                data = json.loads(scan_data)
+            return self.parse_json(data=data, test=test)
         else:
             return ()
 
+    def parse_json(self, data, test):
+        try:
+            vulnerability = data["data"]  # json output of https://pypi.org/project/harborapi/
+        except (KeyError):
+            pass
+        # Early exit if empty
+        if 'data' not in locals() or vulnerability is None:
+            return list()
```

Comment:
How about something like this simplified approach?

```suggestion
        vulnerability = data.get("data")
        if not vulnerability:
            return list()
```